backend/tournesol/views/preview.py

```diff
@@ -260,14 +313,339 @@ def get(self, request, uid):
+        logo_size = (size * self.upscale_ratio, size * self.upscale_ratio)
+        logo = self.load_resource_image("Logo64.png").resize(logo_size)
+        dest = tuple(numpy.multiply(position, self.upscale_ratio))
+        target.alpha_composite(logo, dest=dest)
+
+    def draw_entity_description(self, entity, target, position, available_width):
+        draw = ImageDraw.Draw(target)
+        position = numpy.multiply(position, self.upscale_ratio)
+        available_width *= self.upscale_ratio
+
+        full_uploader = entity.metadata.get("uploader", "")
+        uploader_font = self.font(FOOTER_FONT_LOCATION, 11)
+        truncated_uploader = truncate_text(
+            draw,
+            full_uploader,
+            font=uploader_font,
+            available_width=available_width,
+        )
+        draw.text(
+            position,
+            truncated_uploader,
+            font=uploader_font,
+            fill=COLOR_BROWN_FONT,
+        )
+        uploader_height = font_height(uploader_font)
+
+        full_title = entity.metadata.get("name", "")
+        title_font = self.font(FOOTER_FONT_LOCATION, 14)
+        truncated_title = truncate_text(
+            draw,
+            full_title,
+            font=title_font,
+            available_width=available_width,
+        )
+        draw.text(
+            (position[0], position[1] + uploader_height),
+            truncated_title,
+            font=title_font,
+            fill=COLOR_BROWN_FONT,
+        )
+        title_height = font_height(title_font)
+
+        total_height = (uploader_height + title_height) // self.upscale_ratio
+        return total_height
+
+    def draw_slider(self, target, position):
+        if self.upscale_ratio == 1:
+            comparison_slider = self.load_resource_image("comparison_slider.png")
+        elif self.upscale_ratio == 2:
+            comparison_slider = self.load_resource_image("comparison_slider-2x.png")
+        else:
+            comparison_slider = self.load_resource_image("comparison_slider-4x.png")
+            comparison_slider = comparison_slider.resize(
+                (
+                    target.size[0],
+                    target.size[0] * 160 // 1760,
+                )
+            )
+
+        vertical_margin_in_image = 10
+
+        dest = (
+            position[0] * self.upscale_ratio,
+            (position[1] - vertical_margin_in_image) * self.upscale_ratio,
+        )
+        target.alpha_composite(
+            comparison_slider,
+            dest=dest,
+        )
+        return (
+            comparison_slider.size[1] // self.upscale_ratio
+            - 2 * vertical_margin_in_image
+        )
+
+
+class DynamicWebsitePreviewComparison(BasePreviewAPIView, APIView):
+    """
+    Return the preview of the Tournesol front end's comparison page.
+    """
+
+    permission_classes = []
+
+    # TODO: set the same cache value as DynamicWebsitePreviewEntity
+    @method_decorator(cache_page_no_i18n(0 * 2))
```

Comment:
This file has very slightly evolved on `main` since the beginning of this branch.

Now we have a constant `CACHE_ENTITY_PREVIEW` that we use to define cache of an entity preview. I think we should use this constant too for the comparison preview.

We can merge `main` into this branch, or update this value in the following PR.

Reply:
I merged main and changed this value in c8610e621d8d817a05afd45844e557a45971659f.